### PR TITLE
manix: 0.7.1 -> 0.8.0

### PR DESCRIPTION
**Pull request by IogaMaster**
## Description of changes

@n16hth4wkk and I have moved his maintained fork of manix into nix-community.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by bryango — APPROVED** [review]
This is great! The package builds fine on top of current master.

However, when I try the executable, I got an error:
```console
$ ~/result/bin/manix mergeAttrs
Building Nixpkgs comments cache...
thread '<unnamed>' panicked at src/comments_docsource.rs:194:60:
called `Result::unwrap()` on an `Err` value: UnexpectedWanted(TOKEN_AT, 220..221, [TOKEN_L_PAREN, TOKEN_REC, TOKEN_L_BRACE, TOKEN_L_BRACK, TOKEN_STRING_START, TOKEN_IDENT])
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Is this a packaging issue or a manix issue? Or maybe I am using it wrong?

**Comment by bryango** [comment]
<details><summary>Full back-trace is here.</summary>

```
$ RUST_BACKTRACE=full ~/result/bin/manix mergeAttrs                     
Building Nixpkgs comments cache...
thread '<unnamed>' panicked at src/comments_docsource.rs:194:60:
called `Result::unwrap()` on an `Err` value: UnexpectedWanted(TOKEN_AT, 220..221, [TOKEN_L_PAREN, TOKEN_REC, TOKEN_L_BRACE, TOKEN_L_BRACK, TOKEN_STRING_START, TOKEN_IDENT])
stack backtrace:
   0:     0x555ef67c638b - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h85e32f8f7b08f729
   1:     0x555ef67fd890 - core::fmt::write::hae1cf5cde4ee2188
   2:     0x555ef67bb2d5 - std::io::Write::write_fmt::h4ae54bff47727b29
   3:     0x555ef67c6164 - std::sys_common::backtrace::print::hea248a594c5888cf
   4:     0x555ef67c6db7 - std::panicking::default_hook::{{closure}}::h545cae1ad29f25d2
   5:     0x555ef67c6b04 - std::panicking::default_hook::hc1d7fee5faf0cc8a
   6:     0x555ef67c7318 - std::panicking::rust_panic_with_hook::h0e7975574d129b7c
   7:     0x555ef67c71fe - std::panicking::begin_panic_handler::{{closure}}::h420dcc10a77211ab
   8:     0x555ef67c65a6 - std::sys_common::backtrace::__rust_end_short_backtrace::he6597460302ecd4a
   9:     0x555ef67c6f62 - rust_begin_unwind
  10:     0x555ef66ff025 - core::panicking::panic_fmt::h3cf3338e18ada46e
  11:     0x555ef66fefe3 - core::result::unwrap_failed::h22597ad5fa95281f
  12:     0x555ef6769aaf - <manix::comments_docsource::CommentsDatabase as manix::DocSource>::update::{{closure}}::h88bbede1ba90a798
  13:     0x555ef676977a - <rayon::iter::map::MapFolder<C,F> as rayon::iter::plumbing::Folder<T>>::consume::h8085d629cdb40bbf
  14:     0x555ef6766d22 - rayon::iter::plumbing::bridge_producer_consumer::helper::h4157951f8fb89844
  15:     0x555ef677c0d2 - rayon_core::job::StackJob<L,F,R>::run_inline::he56845543e1811ba
  16:     0x555ef6776c56 - rayon_core::join::join_context::{{closure}}::hda1efde214f4edf5
  17:     0x555ef6766e19 - rayon::iter::plumbing::bridge_producer_consumer::helper::h4157951f8fb89844
  18:     0x555ef6776a86 - rayon_core::join::join_context::{{closure}}::hda1efde214f4edf5
  19:     0x555ef6766e19 - rayon::iter::plumbing::bridge_producer_consumer::helper::h4157951f8fb89844
  20:     0x555ef6776a86 - rayon_core::join::join_context::{{closure}}::hda1efde214f4edf5
  21:     0x555ef6766e19 - rayon::iter::plumbing::bridge_producer_consumer::helper::h4157951f8fb89844
  22:     0x555ef677c0d2 - rayon_core::job::StackJob<L,F,R>::run_inline::he56845543e1811ba
  23:     0x555ef6776c56 - rayon_core::join::join_context::{{closure}}::hda1efde214f4edf5
  24:     0x555ef6766e19 - rayon::iter::plumbing::bridge_producer_consumer::helper::h4157951f8fb89844
  25:     0x555ef6776a86 - rayon_core::join::join_context::{{closure}}::hda1efde214f4edf5
  26:     0x555ef6766e19 - rayon::iter::plumbing::bridge_producer_consumer::helper::h4157951f8fb89844
  27:     0x555ef677e0af - <rayon_core::job::StackJob<L,F,R> as rayon_core::job::Job>::execute::hb6e105b752872627
  28:     0x555ef66faa58 - rayon_core::registry::WorkerThread::wait_until_cold::hc583e99eff65099f
  29:     0x555ef67a7302 - rayon_core::registry::ThreadBuilder::run::h6c241f9408aa5070
  30:     0x555ef67a9e7a - std::sys_common::backtrace::__rust_begin_short_backtrace::h1c1aa5733d7dad65
  31:     0x555ef67ab8ef - core::ops::function::FnOnce::call_once{{vtable.shim}}::hd1024c89ad043895
  32:     0x555ef67e1805 - std::sys::unix::thread::Thread::new::thread_start::h198fdf616b50df6b
  33:     0x7f0d95c07333 - start_thread
  34:     0x7f0d95c89efc - __clone3
  35:                0x0 - <unknown>
```
</details>

**Comment by IogaMaster** [comment]
This is currently an issue with manix.
I don't believe it has anything to do with this pr.

File: pkgs/by-name/ma/manix/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  cargoHash = "sha256-h7K6AAvXrH+k8U5noiIUT6asRMObe50IjW71sCzYnHE=";
+
+  meta = with lib; {
+    description = "A fast CLI documentation searcher for Nix";
+    homepage = "https://github.com/nix-community/manix";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ iogamaster lecoqjacob ];
+    mainProgram = "manix";
+  };
+}
```
Comment (bryango) [inline pkgs/by-name/ma/manix/package.nix:26]:
```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
  pname = "manix";
  version = "0.8.0";

  src = fetchFromGitHub {
    owner = "nix-community";
    repo = "manix";
    rev = "v${finalAttrs.version}";
    hash = "sha256-b/3NvY+puffiQFCQuhRMe81x2wm3vR01MR3iwe/gJkw=";
  };

  cargoHash = "sha256-4qyFVVIlJXgLnkp+Ln4uMlY0BBl8t1na67rSM2iIoEA=";

  meta = with lib; {
    description = "A fast CLI documentation searcher for Nix";
    homepage = "https://github.com/nix-community/manix";
    license = licenses.mpl20;
    maintainers = with maintainers; [ iogamaster lecoqjacob ];
    mainProgram = "manix";
  };
})
```

**Review by bryango — CHANGES_REQUESTED** [review]
The issue has been fixed upstream! :tada: Please bump to v0.8.0. I have suggested the updated hashes below; please double-check. Also, I took the chance to suggest `finalAttrs` which would make it easier to override.

File: pkgs/by-name/ma/manix/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  cargoHash = "sha256-4qyFVVIlJXgLnkp+Ln4uMlY0BBl8t1na67rSM2iIoEA=";
+
+  meta = with lib; {
+    description = "A fast CLI documentation searcher for Nix";
+    homepage = "https://github.com/nix-community/manix";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ iogamaster lecoqjacob ];
+    mainProgram = "manix";
+  };
+})
```
Comment (bryango) [inline pkgs/by-name/ma/manix/package.nix:26]:
```suggestion
rustPlatform.buildRustPackage rec {
  pname = "manix";
  version = "0.8.0";

  src = fetchFromGitHub {
    owner = "nix-community";
    repo = "manix";
    rev = "v${version}";
    hash = "sha256-b/3NvY+puffiQFCQuhRMe81x2wm3vR01MR3iwe/gJkw=";
  };

  cargoHash = "sha256-45cb0yO/ypGLcvEgPOkN6Py99yqK09xnCmMOLOOYYSA=";

  meta = with lib; {
    description = "A fast CLI documentation searcher for Nix";
    homepage = "https://github.com/nix-community/manix";
    license = licenses.mpl20;
    maintainers = with maintainers; [ iogamaster lecoqjacob ];
    mainProgram = "manix";
  };
}
```

**Review by bryango — CHANGES_REQUESTED** [review]
Oh shoot, my bad, so turns out `buildRustPackage` does not support `finalAttrs` yet... Also, I got a hash wrong. This should work now; please check... I am sorry!

**Comment by akarui-hikarii** [comment]
@bryango looks good. I'm not able to push this through since it's @IogaMaster pr. I'll let them review and push through

**Comment by IogaMaster** [comment]
I didn't have time to test before, I can confirm this pr is now fully functional.

**Review by bryango — APPROVED** [review]

**Comment by bryango** [comment]
Result of `nixpkgs-review pr 280617` run on `x86_64-linux`
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>manix</li>
  </ul>
</details>

Result of `nixpkgs-review pr 280617` run on `x86_64-darwin`
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>manix</li>
  </ul>
</details>

**Comment by IogaMaster** [comment]
I just added shell completion and a man page. Once we show docs for the lib functions. Manix will be 1.0.

**Pull request merged**